src/main.rs

```diff
@@ -141,15 +141,54 @@ fn fill_column(uhd: &mut UnicornHatHd, col: usize, colors: Vec<RGB8>) -> Result<
     Ok(())
 }
 
+fn vector_of_leds(vals: Vec<u32>) -> Vec<u64> {
+    let total: u32 = vals.iter().sum();
+
+    if total == 0 {
+        return vec![total as u64];
```

Comment:
Rather than always returning a single-element vec with 0, it'd probably be more general to update the `let ret_val` a little further down to be something like:

```rust
let ret_val = if val == 0 {
    0f64
} else {
    16f64 * (f64::from(val) / f64::from(total))
};
```

That way if all metrics are zero for some reason it'll work if we're passing in more than one number. Eg:

```rust

    let original_vals = vec![0, 0];
    let expected_counts = vec![0, 0];
    let result = vector_of_leds(original_vals.clone());
    assert_eq!(
        result,
        expected_counts,
        "vector_of_leds({:?}) -> {:?} (Expected: {:?})",
        original_vals,
        result,
        expected_counts
    );
```